chapter04/mdw/Ex-3.pl

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/perl
+
+use 5.010;
+
+use strict;
+
+sub average{
+
+  my $total = shift @_;
+  my $count = 1;
+
+  foreach my $number (@_) {
+
+    $total += $number;
+    $count += 1;
+
+  }
+
+$total/$count;
```

Comment:
instead of having this variable you could use: scalar @_ to get the number of elements in the array (provided you didn't shift above)